Addressing issue 89: incorporating SUP on App Registry as OCI Registry (#123)
margo/specification: system-design/concepts/applications/application-package.md

- The **resources**, which are additional files associated with the application (e.g., manual, icon, release notes, license file, etc.) that may be used to provide more information about the application (e.g., in a UI such as an [application catalog](../../personas-and-definitions/technical-lexicon.md#application-catalog) of a [WFM](../../personas-and-definitions/technical-lexicon.md#workload-fleet-manager)).

> The Application Package is distributed via an [Application Registry](./application-registry.md) and is bundled within a **repository**, according to [OCI distribution specification](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md). All files (i.e., Application Description and resource files) belonging to the Application Package are referenced as **layers** by a **manifest** inside the repository. The layer of the Application Description file is marked with a **mediaType** value as "application/vnd.margo.app.description.v1+yaml". The layers of resource files are marked with an **annotation** called "org.margo.app.resource" and a value that indicates the kind of resource (e.g., "metadata.catalog.application.icon"). 

Reviewer: Prefer generic terminology **blob** over the outdated (and still present in some part of the specification for historical reasons) **layer**. The term *layer* is misleading the people to think on container images and OCI artifacts aren't container images, although they rely on the same technology and concepts.

I've done multiple suggestions below to replace that term.

```suggestion
The Application Package is distributed via an [Application Registry](./application-registry.md) and is bundled within a **repository**, according to [OCI distribution specification](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md).
All files (i.e., Application Description and resource files) belonging to the Application Package are referenced as *layers* (the equivalent term **blobs** is nowadays preferred and will be used in the rest of this document) by a **manifest** inside the repository.
The blob of the Application Description file is marked with a **mediaType** value as `application/vnd.margo.app.description.v1+yaml`.
The blobs of resource files are marked with an **annotation** called `org.margo.app.resource` and a value that indicates the kind of resource (e.g., `metadata.catalog.application.icon`). 
```

You are proposing generic mediaTypes to the resources and the use of annotations to declare their use. But I'd rather declare own Margo Media Types (the same [OCI does](https://github.com/opencontainers/image-spec/blob/main/media-types.md)) for our "first-class-citizens" (like probably an application icon). Why because, `mediaType` is an OCI "first-class-citizen" with clear semantics and structure, whereas annotations are sort of the "extension" possibility if that does not suffice.

So I'd rather use `application/vnd.margo.app.icon.v1+png` (or similar, using the `+png` as an option to support other image formats) for the icon and no annotation.

Author: done

This is illustrated in the figure below and formally defined [here](../../specification/applications/application-registry.md).

``` mermaid
classDiagram
direction LR

>     class `my-application-description: Layer` {

Reviewer: ```suggestion
    class `my-application-description: Blob` {
```

Author: done

	    mediaType: "application/vnd.margo.app.description.v1+yaml"
        digest: "sha256:f6b7914..."
        size: 999
    }

>     class `my-resources-icon: Layer` {

Reviewer: ```suggestion
    class `my-resources-icon: Blob` {
```

Author: done

	    mediaType: "image/jpeg"
        digest: "sha256:sha256:451410b6adfdce..."
        size: 5065
        annotations: ["org.margo.app.resource": "metadata.catalog.application.icon", ...]
    }

    class `my-margo-app-manifest: Manifest` {
	    mediaType: "application/vnd.oci.image.manifest.v1+json"
	    artifactType: "application/vnd.org.margo.app.v1+json"
    }

    `my-margo-app-manifest: Manifest` o-- `my-application-description: Layer`

>     `my-margo-app-manifest: Manifest` o-- `my-resources-icon: Layer`

Reviewer: ```suggestion
    `my-margo-app-manifest: Manifest` o-- `my-application-description: Blob`
    `my-margo-app-manifest: Manifest` o-- `my-resources-icon: Blob`
```

Author: done

```

> An application aggregates one or more [components](../../personas-and-definitions/technical-lexicon.md#component), which each link to one or more [OCI Containers](https://github.com/opencontainers). The components  referenced in the [Application Description](../../specification/applications/application-description.md) are stored in a [Component Registry](../../personas-and-definitions/technical-lexicon.md#component-registry), and the linked containers are provided via a [Container Registry](../../personas-and-definitions/technical-lexicon.md#container-registry). Registries can be remote (i.e., Internet-accessible) or [local](../../concepts/applications/local-registries.md) (i.e., accessible within a local network infrastructure of the devices). 

Reviewer: ```suggestion
An application aggregates one or more [components](../../personas-and-definitions/technical-lexicon.md#component), which each link to one or more [OCI Container Images](https://github.com/opencontainers/image-spec).
The components  referenced in the [Application Description](../../specification/applications/application-description.md) are stored in a [Component Registry](../../personas-and-definitions/technical-lexicon.md#component-registry), and the linked container images are provided via a [Container Registry](../../personas-and-definitions/technical-lexicon.md#container-registry).
Registries can be remote (i.e., Internet-accessible) or [local](../../concepts/applications/local-registries.md) (i.e., accessible within a local network infrastructure of the devices). 
```

Author: done

- The **resources**, which are additional files associated with the application (e.g., manual, icon, release notes, license file, etc.) that may be used to display more information about the application in a UI such as an [application catalog](../../personas-and-definitions/technical-lexicon.md#application-catalog) or [marketplace](../../personas-and-definitions/technical-lexicon.md#workload-marketplace) or other informative outputs.

> The Application Package has the following folder/file structure:

Reviewer 2: Is this section still correct? Since we're moving to the OCI blob, we don't have a folder structure anymore, correct?

Reviewer 2: @arne-broering just following up on this comment to see if it was accidentally missed

Author: OK, I removed the folder structure requirement and added an overview of how the OCI repository is bundling the Application Package through a manifest.
